portal/src/main/scala/cromwell/pipeline/service/UserService.scala

```diff
@@ -0,0 +1,21 @@
+package cromwell.pipeline.service
+
+import cromwell.pipeline.datastorage.dao.repository.UserRepository
+import cromwell.pipeline.datastorage.dto.User.UserEmail
+import cromwell.pipeline.datastorage.dto.{ UserId, UserNoCredentials }
+
+import scala.concurrent.{ ExecutionContext, Future }
+
+class UserService(userRepository: UserRepository)(implicit executionContext: ExecutionContext) {
+  def deactivateByEmail(email: UserEmail): Future[Option[UserNoCredentials]] =
+    for {
+      _ <- userRepository.deactivateByEmail(email)
```

Comment:
Why not refactor the `deactivateByEmail` method to make it return a deactivated user? Same for the `deactivateById` method.